### PR TITLE
add ports to inspect output

### PR DESCRIPTION
**Pull request by fahedouch**
fixing https://github.com/containerd/nerdctl/issues/130#issue-837210352

Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,6 +27,7 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/rootless-containers/rootlesskit v0.14.1
 	github.com/sirupsen/logrus v1.8.1
+	github.com/stretchr/testify v1.7.0
```
Comment (AkihiroSuda) [inline go.mod:30]: Unneeded dependency

File: pkg/inspecttypes/dockercompat/dockercompat.go [hunk 2]
```diff
@@ -214,9 +220,18 @@ func networkSettingsFromNative(n *native.NetNS) *NetworkSettings {
 		// TODO: set CNI name when possible
 		fakeDockerNetworkName := fmt.Sprintf("unknown-%s", x.Name)
 		res.Networks[fakeDockerNetworkName] = nes
+
+		sp := l.Spec.(*specs.Spec).Annotations[labels.Ports]
```
Comment (AkihiroSuda) [inline pkg/inspecttypes/dockercompat/dockercompat.go:224]: Unneeded assertion

File: pkg/inspecttypes/dockercompat/dockercompat.go [hunk 3]
```diff
@@ -214,9 +220,18 @@ func networkSettingsFromNative(n *native.NetNS) *NetworkSettings {
...
+
+		var ports []gocni.PortMapping
+		json.Unmarshal([]byte(sp), &ports)
```
Comment (AkihiroSuda) [inline pkg/inspecttypes/dockercompat/dockercompat.go:227]: Check err

File: container_inspect_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"github.com/stretchr/testify/assert"
```
Comment (AkihiroSuda) [inline container_inspect_test.go:20]: Please use existing assert package

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: container_inspect_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+func TestContainerInspectContainsPortConfig(t *testing.T) {
+	const (
+		testContainer0 = "nerdctl-test-inspect-with-port-config"
+	)
+
+	base := testutil.NewBase(t)
+	defer base.Cmd("rm", "-f", testContainer0).Run()
+
+	const expected = `{"80/tcp":[{"HostIp":"0.0.0.0","HostPort":"80"}]}`
+	base.Cmd("run", "-d", "--name", testContainer0, "-p", "80:80", testutil.NginxAlpineImage).Run()
```
Comment (AkihiroSuda) [inline container_inspect_test.go:36]: Please use unprivileged port, and add AssertOK

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: pkg/inspecttypes/dockercompat/dockercompat.go [hunk 6]
```diff
@@ -214,9 +224,21 @@ func networkSettingsFromNative(n *native.NetNS) *NetworkSettings {
 		// TODO: set CNI name when possible
 		fakeDockerNetworkName := fmt.Sprintf("unknown-%s", x.Name)
 		res.Networks[fakeDockerNetworkName] = nes
+
+		var ports []gocni.PortMapping
+		err := json.Unmarshal([]byte(sp.Annotations[labels.Ports]), &ports)
```
Comment (AkihiroSuda) [inline pkg/inspecttypes/dockercompat/dockercompat.go:229]: Skip setting `res.Ports` when `if portsLabel, ok := sp.Annotations[labels.Ports]; !ok`

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: pkg/inspecttypes/dockercompat/dockercompat.go (hunk 6 above)
Comment (fahedouch) [reply pkg/inspecttypes/dockercompat/dockercompat.go]: done with omit the empty json. is it for you ?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**